addon/adapter.go

```diff
@@ -135,14 +135,14 @@ func newAdapter() (adapter *Adapter) {
 		Tag: Tag{
 			client: client,
 		},
-		File: File{
-			client: client,
-		},
-		RuleSet: RuleSet{
-			client: client,
-		},
-		client: client,
+		File:    binding.File{},
+		RuleSet: binding.RuleSet{},
+		client:  client,
 	}
+	//
+	// Setting client in binding package via setter method.
+	adapter.File.SetClient(client)
+	adapter.RuleSet.SetClient(client)
```

Comment:
Since ```client``` is ot available outside the package (binding in this case), I added SetClient methods to allow set the client from addon package. Expecting its deletion when addon is migrated to binding.

Reply:
In the binding package, we could just make the `client` in handlers_public_ instead?

Reply:
Sure, we can, will update.